apps/web/app/api/teams/[team]/upgrade/route.ts

```diff
@@ -0,0 +1,104 @@
+import { cookies, headers } from "next/headers";
+import type { NextRequest } from "next/server";
+import { NextResponse } from "next/server";
+import type Stripe from "stripe";
+import { z } from "zod";
+
+import { getRequestedSlugError } from "@calcom/app-store/stripepayment/lib/team-billing";
+import { getServerSession } from "@calcom/features/auth/lib/getServerSession";
+import stripe from "@calcom/features/ee/payments/server/stripe";
+import { WEBAPP_URL } from "@calcom/lib/constants";
+import { HttpError } from "@calcom/lib/http-error";
+import prisma from "@calcom/prisma";
+import { teamMetadataSchema } from "@calcom/prisma/zod-utils";
+
+import { buildLegacyRequest } from "@lib/buildLegacyCtx";
+
+const querySchema = z.object({
+  team: z.string().transform((val) => parseInt(val)),
+  session_id: z.string().min(1),
+});
+
+export async function GET(req: NextRequest, { params }: { params: { team: string } }) {
+  try {
+    const searchParams = req.nextUrl.searchParams;
+    const { team: id, session_id } = querySchema.parse({
+      team: params.team,
+      session_id: searchParams.get("session_id"),
+    });
```

Comment:
replacing `const { team: id, session_id } = querySchema.parse(req.query);` in the pages router api file